### PR TITLE
basic: Added sub command

### PR DESCRIPTION
**Pull request by mrshu**
* Added sub command which encapsulates 'sed substitution' in a command.

* Moved the regular expression which matches the sed substitution
  syntax into a separate variable.

* Did some janitoring throughout the file.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

**Comment by mrshu** [comment]
to @pepol 

cc @JakubNvk, @Adman

File: basic.py [hunk 1]
```diff
@@ -103,11 +104,11 @@ def sub_event_catcher(event):
         last_events[event.meta['host']] = event
 
 
-@match(regex=r'^s([^ a-zA-Z0-9])(.*?)\1(.*?)(?:\1(.+?))?$')
+@match(regex=SED_REGEX)
 def sub_match(event, sep, pattern, replacement, flags, *args, **kwargs):
     """Matches any substitute string (as sed would), applies this substitution to
-    the user's last message and sents it back to processing.
-    """
+    the user's last message and sends it back to processing."""
```
Comment (pepol) [inline basic.py]: @mrshu actually, as per [PEP257](https://www.python.org/dev/peps/pep-0257) (Docstring conventions):
"Unless the entire docstring fits on a line, place the closing quotes on a line by themselves."

**Comment by pepol** [comment]
@mrshu apart from that one note about docstrings (but there is a question how we want to deal with this), looks good.

File: basic.py (hunk 1 above)
Comment (mrshu) [inline basic.py]: @pepol yeah, but you have omitted the second part

``` This way, Emacs' fill-paragraph command can be used on it. ```

If that's the reason for that I am very much OK with leaving it this way here.

Comment (pepol) [inline basic.py]: Then go ahead, you have my blessing :D

**Comment by pepol** [comment]
LGTM

**Comment by mrshu** [comment]
Thanks @pepol !

**Pull request merged**